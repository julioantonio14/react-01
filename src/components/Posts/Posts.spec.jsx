import { fireEvent,render,screen } from "@testing-library/react"
import {Posts} from '.';

const props = {
    posts:[{
        id:1,
        title: 'Titulo 1',
        body:'corpo 1',
        cover: 'img/img1.png'   
    },{
        id:2,
        title: 'Titulo 2',
        body:'corpo 2',
        cover: 'img/img2.png'   
    },{
        id:3,
        title: 'Titulo 3',
        body:'corpo 3',
        cover: 'img/img3.png'   
    },]
}

describe('<Posts />', ()=>{
    it('should render posts', ()=>{ 
        render(<Posts {...props}/>)
        expect(screen.getAllByRole('heading', {name: /Titulo/i})).toHaveLength(3)
        expect(screen.getAllByRole('img', {name: /Titulo/i})).toHaveLength(3)
        expect(screen.getAllByText(/corpo/i)).toHaveLength(3)
        expect(screen.getByRole('img', {name: /Titulo 3/i})).toHaveAttribute('src', 'img/img3.png')
    })
    it('should not render posts', ()=>{ 
        render(<Posts/>);
        expect(screen.queryAllByRole('heading', {name: /Titulo/i})).toHaveLength(0)
    })
    it('should match snapshot', ()=>{ 
        const {container} = render(<Posts {...props}/>);
        expect(container.firstChild).toMatchSnapshot();
    })
})