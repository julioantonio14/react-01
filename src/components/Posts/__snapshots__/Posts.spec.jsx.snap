// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Titulo 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Titulo 1
      </h2>
      <p>
        corpo 1
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Titulo 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Titulo 2
      </h2>
      <p>
        corpo 2
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Titulo 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Titulo 3
      </h2>
      <p>
        corpo 3
      </p>
      <p>
        Oi
      </p>
    </div>
  </div>
</div>
`;
